CombinatorialGames/Surreal/Cut.lean

```diff
@@ -149,16 +247,298 @@ theorem mem_right_rightSurreal {x y} : y ∈ (rightSurreal x).right ↔ x < y :=
+
+/-- The simplest surreal number (in terms of birthday) lying between two cuts. -/
+noncomputable def simplestBtwn {x y : Cut} (h : x < y) : Surreal :=
+  Classical.choose <|
+    exists_minimalFor_of_wellFoundedLT _ birthday (lt_iff_nonempty_inter.1 h)
+
+private theorem simplestBtwn_spec {x y : Cut} (h : x < y) :
+    MinimalFor (fun z ↦ z ∈ x.right ∩ y.left) birthday (simplestBtwn h) :=
+  Classical.choose_spec _
+
+theorem simplestBtwn_mem {x y : Cut} (h : x < y) : simplestBtwn h ∈ x.right ∩ y.left :=
+  (simplestBtwn_spec h).1
+
+theorem birthday_simplestBtwn_le_of_mem {x y : Cut} {z : Surreal} (h : x < y)
+    (hz : z ∈ x.right ∩ y.left) : (simplestBtwn h).birthday ≤ z.birthday := by
+  by_contra! H
+  exact H.not_ge <| (simplestBtwn_spec h).2 hz H.le
+
+theorem equiv_of_mem_supLeft_inter_infRight {x y : IGame} [y.Numeric]
+    (hy : .mk y ∈ (supLeft x).right ∩ (infRight x).left)
+    (hol : ∀ z (h : z ∈ y.leftMoves),
+      have := Numeric.of_mem_leftMoves h; .mk z ∈ (supLeft x).left)
+    (hor : ∀ z (h : z ∈ y.rightMoves),
+      have := Numeric.of_mem_rightMoves h; .mk z ∈ (infRight x).right) :
+    x ≈ y := by
+  unfold supLeft infRight at *
+  constructor <;> refine le_iff_forall_lf.2 ⟨?_, ?_⟩ <;> intro z hz
+  · simp_all
+  · simp_rw [right_iInf, mem_iUnion] at hor
+    obtain ⟨i, hi, hor⟩ := hor z hz
+    refine lf_of_rightMove_le ?_ hi
+    simpa
+  · simp_rw [left_iSup, mem_iUnion] at hol
+    obtain ⟨i, hi, hol⟩ := hol z hz
+    refine lf_of_le_leftMove ?_ hi
+    simpa
+  · simp_all
+
+theorem simplestBtwn_leftGame_rightGame {x : Game} (h : leftGame x < rightGame x) :
+    (simplestBtwn h).toGame = x := by
+  rw [leftGame_lt_rightGame_iff] at h
+  obtain ⟨x, rfl⟩ := h
+  have hs := simplestBtwn_mem h
+  simp_all [le_antisymm_iff]
+
+@[simp]
+theorem simplestBtwn_leftSurreal_rightSurreal (x : Surreal) :
+    simplestBtwn (leftSurreal_lt_rightSurreal x) = x := by
+  convert simplestBtwn_leftGame_rightGame (x := x.toGame) _ <;> simp
+
+theorem simplestBtwn_supLeft_infRight {x : IGame} (h : supLeft x < infRight x) :
+    (simplestBtwn h).toGame = .mk x := by
+  have H := simplestBtwn_mem h
+  obtain ⟨y, _, hy, hy'⟩ := birthday_eq_iGameBirthday (simplestBtwn h)
+  rw [← hy, toGame_mk, Game.mk_eq_mk]
+  apply (equiv_of_mem_supLeft_inter_infRight ..).symm
+  · simp_all
+  · rw [← compl_right]
+    intro z hz _ hz'
+    apply (hy' ▸ birthday_lt_of_mem_leftMoves hz).not_ge
+    apply (birthday_simplestBtwn_le_of_mem ..).trans (birthday_mk_le z)
+    refine ⟨hz', isLowerSet_left _ (mk_lt_mk.2 <| Numeric.leftMove_lt hz).le ?_⟩
+    aesop
+  · rw [← compl_left]
+    intro z hz _ hz'
+    apply (hy' ▸ birthday_lt_of_mem_rightMoves hz).not_ge
+    apply (birthday_simplestBtwn_le_of_mem ..).trans (birthday_mk_le z)
+    refine ⟨isUpperSet_right _ (mk_lt_mk.2 <| Numeric.lt_rightMove hz).le ?_, hz'⟩
+    aesop
+
+/-! ### Small cuts -/
+
+/-- A "small cut" is defined as either the infimum of a small set of left cuts of surreals,
```

Comment:
Should this be a class? It doesn't help here, but it would be consistent how this is an inductively defined proposition whose prefixed with `Is`

```suggestion
class inductive IsSmall : Cut.{u} → Prop
  | sInf' (s : Set Surreal) [Small.{u} s] : (sInf (leftSurreal '' s)).IsSmall
  | sSup' (s : Set Surreal) [Small.{u} s] : (sSup (rightSurreal '' s)).IsSmall
```